expression/collation.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/tidb/types"
+)
+
+type coercibility struct {
+	value Coercibility
+	init  bool
+}
+
+func (c *coercibility) hasCoercibility() bool {
+	return c.init
+}
+
+func (c *coercibility) coercibility() Coercibility {
```

Comment:
How about renaming it to `value()`? So the usage of it can be something like:

```
...
return sf.coercibility.value()
```
, which looks better.